server/src/server.js

```diff
@@ -743,6 +743,30 @@ app.post("/api/set-title/:id/:domain", csrfProtection, function(req, res) {
   });
 });
 
+app.post("/api/save-edit", csrfProtection, function(req, res) {
+  let vars = req.body;
+  if (!req.deviceId) {
+    sendRavenMessage(req, "Attempt to set expiration without login");
+    simpleResponse(res, "Not logged in", 401);
+    return;
+  }
+  let id = vars.shotId;
+  let url = vars.url;
+  Shot.get(req.backend, id).then((shot) => {
```

Comment:
There's no permission check here. There should be a deviceId/accountId permission check as part of `Shot.get()`